### DISCUSSION TITLE
No_PROPOSAL_CHOSEN notify error im not an programmer.

### DISCUSSION THREAD

**Discussion by killgapwalker** [strongswan/strongswan#351] [General]
initiating IKE_SA NordVPN[1] to 196.196.216.20
generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
sending packet: from 192.168.1.100[500] to 196.196.216.20[500] (1492 bytes)
received packet: from 196.196.216.20[500] to 192.168.1.100[500] (36 bytes)
parsed IKE_SA_INIT response 0 [ N(NO_PROP) ]
received NO_PROPOSAL_CHOSEN notify error
establishing connection 'NordVPN' failed

Config setup
        charondebug="all"
        uniqueids=yes
conn NordVPN
        keyexchange=ikev2
        dpdaction=clear
        dpddelay=300
        eap_identity=************************
        leftauth=eap
        left=192.168.1.100
        leftsubnet=192.168.1.100/24
        right=196.196.216.20
        rightsubnet=196.196.216.20/24
        rightca=/etc/ipsec.d/cacerts/NordVPN.pem
        type=tunnel
        auto=add
        fragmentation=yes
        ike=aes128-sha256-modp3072
        esp=aes128-sha256-modp3072
        leftsubnet=%dynamic
        mark=%unique
        rightauth2=xauth-generic
        xauth=server

**Comment by Thermi** [comment]
That doesn't fit forwards OR backwards.

You specify ikev2 and then leftauth eap, without a method, and then continue with a nonsense config with nonsense left and rightsubnet and then specify leftsubnet=%dynamic and mark=%unique and rightauth2=xauth-generic. None of that fits in any way.

What are you even trying to accomplish?!

**Comment by killgapwalker** [comment]
I'm trying to connect notebook client in NordVPN servers,
From mobile network.

сб, 24 апр. 2021 г., 21:55 Thermi ***@***.***>:

**Comment by killgapwalker** [comment]
SAME WITHOUT SWITCHES IF IT GOES LIKE MISTAKE

initiating IKE_SA NordVPN[1] to 196.196.216.20
generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
sending packet: from 192.168.1.100[500] to 196.196.216.20[500] (1492 bytes)
received packet: from 196.196.216.20[500] to 192.168.1.100[500] (36 bytes)
parsed IKE_SA_INIT response 0 [ N(NO_PROP) ]
received NO_PROPOSAL_CHOSEN notify error
establishing connection 'NordVPN' failed

Config setup
charondebug="all"
uniqueids=yes
conn NordVPN
keyexchange=ikev2
dpdaction=clear
dpddelay=300
eap_identity=************************
leftauth=eap
right=196.196.216.20
rightca=/etc/ipsec.d/cacerts/NordVPN.pem
type=tunnel
auto=add
fragmentation=yes
ike=aes128-sha256-modp3072
esp=aes128-sha256-modp3072
leftsubnet=%dynamic
mark=%unique
rightauth2=xauth-generic
xauth=server

**Reply by killgapwalker** [reply]
and without mark,rightauth2,xauth is same problem... somebody understand 
/cat/log/syslog/
5 08:04:20 python-Aspire-5737Z wpa_supplicant[751]: message repeated 22 times: [ wlp17s0: CTRL-EVENT-BEACON-LOSS ]
Apr 25 08:04:21 python-Aspire-5737Z charon: 08[CFG] received stroke: initiate 'NordVPN'
Apr 25 08:04:21 python-Aspire-5737Z charon: 13[IKE] initiating IKE_SA NordVPN[2] to 196.196.216.20
Apr 25 08:04:21 python-Aspire-5737Z charon: 13[ENC] generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
Apr 25 08:04:21 python-Aspire-5737Z charon: 13[NET] sending packet: from 192.168.43.221[500] to 196.196.216.20[500] (1492 bytes)
Apr 25 08:04:21 python-Aspire-5737Z charon: 14[NET] received packet: from 196.196.216.20[500] to 192.168.43.221[500] (36 bytes)
Apr 25 08:04:21 python-Aspire-5737Z charon: 14[ENC] parsed IKE_SA_INIT response 0 [ N(NO_PROP) ]
Apr 25 08:04:21 python-Aspire-5737Z charon: 14[IKE] received NO_PROPOSAL_CHOSEN notify error
Apr 25 08:04:23 python-Aspire-5737Z wpa_supplicant[751]: wlp17s0: CTRL-EVENT-BEACON-LOSS

**Reply by Thermi** [reply]
The parameters you changed are not relevant for the initial negotiation.
I assume at the very least the IKE cipher proposal needs to be changed.
Try ike=aes192-sha1-mod2048

**Reply by Thermi** [reply]
Following https://support.nordvpn.com/FAQ/1047408592/Which-protocol-should-I-choose.htm:
```
The ciphers used to generate Phase1 keys are AES-256-GCM for encryption, coupled with SHA2-384 to ensure integrity, and combined with PFS (Perfect Forward Secrecy) using 3072-bit Diffie-Hellman keys.
```
It could also be ike=aes256gcm16-sha384-modp3072

**Comment by killgapwalker** [comment]
Apr 25 11:15:01 python-Aspire-5737Z charon: 00[DMN] signal of type SIGINT received. Shutting down
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[DMN] Starting IKE charon daemon (strongSwan 5.8.4, Linux 5.8.0-25-generic, x86_64)
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[CFG] PKCS11 module '<name>' lacks library path
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[CFG] loaded 0 RADIUS server configurations
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[CFG] HA config misses local/remote address
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[CFG] loading ca certificates from '/etc/ipsec.d/cacerts'
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[CFG]   loaded ca certificate "C=PA, O=NordVPN, CN=NordVPN Root CA" from '/etc/ipsec.d/cacerts/NordVPN.der'
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[CFG]   loaded ca certificate "C=PA, O=NordVPN, CN=NordVPN Root CA" from '/etc/ipsec.d/cacerts/NordVPN.pem'
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[CFG] loading aa certificates from '/etc/ipsec.d/aacerts'
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[CFG] loading ocsp signer certificates from '/etc/ipsec.d/ocspcerts'
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[CFG] loading attribute certificates from '/etc/ipsec.d/acerts'
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[CFG] loading crls from '/etc/ipsec.d/crls'
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[CFG] loading secrets from '/etc/ipsec.secrets'
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[CFG]   loaded EAP secret for H92zYzBsLeGjnyn5KjYWTyxy
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[LIB] loaded plugins: charon addrblock af-alg agent attr ccm certexpire chapoly cmac connmark counters ctr dhcp dnskey drbg eap-dynamic eap-gtc eap-md5 eap-mschapv2 eap-peap eap-radius eap-tls eap-tnc eap-ttls error-notify farp gcm gcrypt ldap led lookip md5 mgf1 ntru openssl pgp pkcs11 pkcs12 pkcs7 pkcs8 pubkey random rc2 resolve sshkey test-vectors tnc-tnccs tpm unity xauth-eap xauth-generic xauth-pam xcbc nonce aes sha1 sha2 pem pkcs1 curve25519 gmp x509 curl revocation hmac stroke kernel-netlink socket-default fips-prf eap-aka eap-identity updown
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[LIB] dropped capabilities, running as uid 0, gid 0
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[JOB] spawning 16 worker threads
Apr 25 11:15:03 python-Aspire-5737Z charon: 05[CFG] received stroke: add connection 'NordVPN'
Apr 25 11:15:03 python-Aspire-5737Z charon: 05[CFG] CA certificate "/etc/ipsec.d/cacerts/NordVPN.pem" not found, discarding CA constraint
Apr 25 11:15:03 python-Aspire-5737Z charon: 05[CFG] added configuration 'NordVPN'
Apr 25 11:15:11 python-Aspire-5737Z charon: 07[CFG] received stroke: initiate 'NordVPN'
Apr 25 11:15:11 python-Aspire-5737Z charon: 09[IKE] initiating IKE_SA NordVPN[1] to 196.196.216.20
Apr 25 11:15:11 python-Aspire-5737Z charon: 09[ENC] generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
Apr 25 11:15:11 python-Aspire-5737Z charon: 09[NET] sending packet: from 192.168.43.221[500] to 196.196.216.20[500] (1364 bytes)
Apr 25 11:15:11 python-Aspire-5737Z charon: 10[NET] received packet: from 196.196.216.20[500] to 192.168.43.221[500] (36 bytes)
Apr 25 11:15:11 python-Aspire-5737Z charon: 10[ENC] parsed IKE_SA_INIT response 0 [ N(NO_PROP) ]
Apr 25 11:15:11 python-Aspire-5737Z charon: 10[IKE] received NO_PROPOSAL_CHOSEN notify error
Apr 25 11:16:24 python-Aspire-5737Z rtkit-daemon[886]: Supervising 4 threads of 2 processes of 1 users.

**Reply by killgapwalker** [reply]
if im right , problem in certificate ?

**Reply by Thermi** [reply]
No. The certificate is not exchanged yet.

**Comment by killgapwalker** [comment]
CHANGED CERTIFICATE....
Apr 25 11:15:01 python-Aspire-5737Z charon: 00[DMN] signal of type SIGINT received. Shutting down
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[DMN] Starting IKE charon daemon (strongSwan 5.8.4, Linux 5.8.0-25-generic, x86_64)
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[CFG] PKCS11 module '<name>' lacks library path
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[CFG] loaded 0 RADIUS server configurations
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[CFG] HA config misses local/remote address
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[CFG] loading ca certificates from '/etc/ipsec.d/cacerts'
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[CFG]   loaded ca certificate "C=PA, O=NordVPN, CN=NordVPN Root CA" from '/etc/ipsec.d/cacerts/NordVPN.der'
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[CFG]   loaded ca certificate "C=PA, O=NordVPN, CN=NordVPN Root CA" from '/etc/ipsec.d/cacerts/NordVPN.pem'
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[CFG] loading aa certificates from '/etc/ipsec.d/aacerts'
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[CFG] loading ocsp signer certificates from '/etc/ipsec.d/ocspcerts'
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[CFG] loading attribute certificates from '/etc/ipsec.d/acerts'
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[CFG] loading crls from '/etc/ipsec.d/crls'
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[CFG] loading secrets from '/etc/ipsec.secrets'
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[CFG]   loaded EAP secret for H92zYzBsLeGjnyn5KjYWTyxy
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[LIB] loaded plugins: charon addrblock af-alg agent attr ccm certexpire chapoly cmac connmark counters ctr dhcp dnskey drbg eap-dynamic eap-gtc eap-md5 eap-mschapv2 eap-peap eap-radius eap-tls eap-tnc eap-ttls error-notify farp gcm gcrypt ldap led lookip md5 mgf1 ntru openssl pgp pkcs11 pkcs12 pkcs7 pkcs8 pubkey random rc2 resolve sshkey test-vectors tnc-tnccs tpm unity xauth-eap xauth-generic xauth-pam xcbc nonce aes sha1 sha2 pem pkcs1 curve25519 gmp x509 curl revocation hmac stroke kernel-netlink socket-default fips-prf eap-aka eap-identity updown
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[LIB] dropped capabilities, running as uid 0, gid 0
Apr 25 11:15:03 python-Aspire-5737Z charon: 00[JOB] spawning 16 worker threads
Apr 25 11:15:03 python-Aspire-5737Z charon: 05[CFG] received stroke: add connection 'NordVPN'
Apr 25 11:15:03 python-Aspire-5737Z charon: 05[CFG] CA certificate "/etc/ipsec.d/cacerts/NordVPN.pem" not found, discarding CA constraint
Apr 25 11:15:03 python-Aspire-5737Z charon: 05[CFG] added configuration 'NordVPN'
Apr 25 11:15:11 python-Aspire-5737Z charon: 07[CFG] received stroke: initiate 'NordVPN'
Apr 25 11:15:11 python-Aspire-5737Z charon: 09[IKE] initiating IKE_SA NordVPN[1] to 196.196.216.20
Apr 25 11:15:11 python-Aspire-5737Z charon: 09[ENC] generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
Apr 25 11:15:11 python-Aspire-5737Z charon: 09[NET] sending packet: from 192.168.43.221[500] to 196.196.216.20[500] (1364 bytes)
Apr 25 11:15:11 python-Aspire-5737Z charon: 10[NET] received packet: from 196.196.216.20[500] to 192.168.43.221[500] (36 bytes)
Apr 25 11:15:11 python-Aspire-5737Z charon: 10[ENC] parsed IKE_SA_INIT response 0 [ N(NO_PROP) ]
Apr 25 11:15:11 python-Aspire-5737Z charon: 10[IKE] received NO_PROPOSAL_CHOSEN notify error
Apr 25 11:16:24 python-Aspire-5737Z rtkit-daemon[886]: Supervising 4 threads of 2 processes of 1 users.
Apr 25 11:16:24 python-Aspire-5737Z rtkit-daemon[886]: Supervising 4 threads of 2 processes of 1 users.
Apr 25 11:17:01 python-Aspire-5737Z CRON[2776]: (root) CMD (   cd / && run-parts --report /etc/cron.hourly)
Apr 25 11:18:16 python-Aspire-5737Z systemd-resolved[673]: Using degraded feature set UDP instead of UDP+EDNS0 for DNS server 192.168.43.1.
Apr 25 11:21:53 python-Aspire-5737Z dbus-daemon[1534]: [session uid=1000 pid=1534] Activating service name='org.gnome.Nautilus' requested by ':1.38' (uid=1000 pid=1759 comm="/usr/bin/gnome-shell " label="unconfined")
Apr 25 11:21:53 python-Aspire-5737Z dbus-daemon[1534]: [session uid=1000 pid=1534] Successfully activated service 'org.gnome.Nautilus'
Apr 25 11:21:54 python-Aspire-5737Z dbus-daemon[716]: [system] Activating via systemd: service name='org.freedesktop.hostname1' unit='dbus-org.freedesktop.hostname1.service' requested by ':1.117' (uid=1000 pid=2804 comm="/usr/bin/nautilus --gapplication-service " label="unconfined")
Apr 25 11:21:54 python-Aspire-5737Z systemd[1]: Starting Hostname Service...
Apr 25 11:21:54 python-Aspire-5737Z dbus-daemon[716]: [system] Successfully activated service 'org.freedesktop.hostname1'
Apr 25 11:21:54 python-Aspire-5737Z systemd[1]: Started Hostname Service.
Apr 25 11:21:55 python-Aspire-5737Z nautilus[2804]: Called "net usershare info" but it failed: Failed to execute child process “net” (No such file or directory)
Apr 25 11:21:56 python-Aspire-5737Z wpa_supplicant[758]: wlp17s0: CTRL-EVENT-BEACON-LOSS 
Apr 25 11:22:22 python-Aspire-5737Z wpa_supplicant[758]: message repeated 12 times: [ wlp17s0: CTRL-EVENT-BEACON-LOSS ]
Apr 25 11:22:24 python-Aspire-5737Z systemd[1]: systemd-hostnamed.service: Succeeded.
Apr 25 11:22:26 python-Aspire-5737Z wpa_supplicant[758]: wlp17s0: CTRL-EVENT-BEACON-LOSS 
Apr 25 11:23:04 python-Aspire-5737Z wpa_supplicant[758]: message repeated 12 times: [ wlp17s0: CTRL-EVENT-BEACON-LOSS ]
Apr 25 11:23:19 python-Aspire-5737Z rtkit-daemon[886]: Supervising 4 threads of 2 processes of 1 users.
Apr 25 11:23:19 python-Aspire-5737Z rtkit-daemon[886]: Supervising 4 threads of 2 processes of 1 users.
Apr 25 11:23:24 python-Aspire-5737Z wpa_supplicant[758]: wlp17s0: CTRL-EVENT-BEACON-LOSS 
Apr 25 11:23:35 python-Aspire-5737Z wpa_supplicant[758]: wlp17s0: CTRL-EVENT-BEACON-LOSS 
Apr 25 11:23:40 python-Aspire-5737Z rtkit-daemon[886]: Supervising 4 threads of 2 processes of 1 users.
Apr 25 11:23:40 python-Aspire-5737Z rtkit-daemon[886]: Supervising 4 threads of 2 processes of 1 users.
Apr 25 11:23:44 python-Aspire-5737Z wpa_supplicant[758]: wlp17s0: CTRL-EVENT-BEACON-LOSS 
Apr 25 11:24:58 python-Aspire-5737Z wpa_supplicant[758]: message repeated 8 times: [ wlp17s0: CTRL-EVENT-BEACON-LOSS ]
Apr 25 11:26:08 python-Aspire-5737Z rtkit-daemon[886]: Supervising 4 threads of 2 processes of 1 users.
Apr 25 11:26:08 python-Aspire-5737Z rtkit-daemon[886]: Supervising 4 threads of 2 processes of 1 users.
Apr 25 11:29:15 python-Aspire-5737Z systemd[1520]: gnome-terminal-server.service: Succeeded.
Apr 25 11:30:01 python-Aspire-5737Z CRON[2952]: (root) CMD ([ -x /etc/init.d/anacron ] && if [ ! -d /run/systemd/system ]; then /usr/sbin/invoke-rc.d anacron start >/dev/null; fi)
Apr 25 11:30:13 python-Aspire-5737Z dbus-daemon[1534]: [session uid=1000 pid=1534] Activating via systemd: service name='org.gnome.Terminal' unit='gnome-terminal-server.service' requested by ':1.97' (uid=1000 pid=2954 comm="/usr/bin/gnome-terminal.real --wait " label="unconfined")
Apr 25 11:30:13 python-Aspire-5737Z systemd[1520]: Starting GNOME Terminal Server...
Apr 25 11:30:13 python-Aspire-5737Z dbus-daemon[1534]: [session uid=1000 pid=1534] Successfully activated service 'org.gnome.Terminal'
Apr 25 11:30:13 python-Aspire-5737Z systemd[1520]: Started GNOME Terminal Server.
Apr 25 11:30:14 python-Aspire-5737Z systemd[1520]: Started VTE child process 2965 launched by gnome-terminal-server process 2957.
Apr 25 11:30:19 python-Aspire-5737Z systemd[1]: Started Run anacron jobs.
Apr 25 11:30:19 python-Aspire-5737Z anacron[2973]: Anacron 2.3 started on 2021-04-25
Apr 25 11:30:19 python-Aspire-5737Z anacron[2973]: Normal exit (0 jobs run)
Apr 25 11:30:19 python-Aspire-5737Z systemd[1]: anacron.service: Succeeded.
Apr 25 11:30:57 python-Aspire-5737Z charon: 14[CFG] received stroke: initiate 'NordVPN'
Apr 25 11:30:57 python-Aspire-5737Z charon: 16[IKE] initiating IKE_SA NordVPN[2] to 196.196.216.20
Apr 25 11:30:57 python-Aspire-5737Z charon: 16[ENC] generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
Apr 25 11:30:57 python-Aspire-5737Z charon: 16[NET] sending packet: from 192.168.43.221[500] to 196.196.216.20[500] (1364 bytes)
Apr 25 11:30:57 python-Aspire-5737Z charon: 06[NET] received packet: from 196.196.216.20[500] to 192.168.43.221[500] (36 bytes)
Apr 25 11:30:57 python-Aspire-5737Z charon: 06[ENC] parsed IKE_SA_INIT response 0 [ N(NO_PROP) ]
Apr 25 11:30:57 python-Aspire-5737Z charon: 06[IKE] received NO_PROPOSAL_CHOSEN notify error

**Comment by killgapwalker** [comment]
Apr 25 11:30:13 python-Aspire-5737Z systemd[1520]: Started GNOME Terminal Server.
Apr 25 11:30:14 python-Aspire-5737Z systemd[1520]: Started VTE child process 2965 launched by gnome-terminal-server process 2957.
Apr 25 11:30:19 python-Aspire-5737Z systemd[1]: Started Run anacron jobs.
Apr 25 11:30:19 python-Aspire-5737Z anacron[2973]: Anacron 2.3 started on 2021-04-25
Apr 25 11:30:19 python-Aspire-5737Z anacron[2973]: Normal exit (0 jobs run)
Apr 25 11:30:19 python-Aspire-5737Z systemd[1]: anacron.service: Succeeded.
Apr 25 11:30:57 python-Aspire-5737Z charon: 14[CFG] received stroke: initiate 'NordVPN'
Apr 25 11:30:57 python-Aspire-5737Z charon: 16[IKE] initiating IKE_SA NordVPN[2] to 196.196.216.20
Apr 25 11:30:57 python-Aspire-5737Z charon: 16[ENC] generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
Apr 25 11:30:57 python-Aspire-5737Z charon: 16[NET] sending packet: from 192.168.43.221[500] to 196.196.216.20[500] (1364 bytes)
Apr 25 11:30:57 python-Aspire-5737Z charon: 06[NET] received packet: from 196.196.216.20[500] to 192.168.43.221[500] (36 bytes)
Apr 25 11:30:57 python-Aspire-5737Z charon: 06[ENC] parsed IKE_SA_INIT response 0 [ N(NO_PROP) ]
Apr 25 11:30:57 python-Aspire-5737Z charon: 06[IKE] received NO_PROPOSAL_CHOSEN notify error
Apr 25 11:36:25 python-Aspire-5737Z /usr/libexec/gdm-x-session[1590]: (EE) event7  - COMPANY 2.4G Device Mouse: client bug: event processing lagging behind by 15ms, your system is too slow
Apr 25 11:39:06 python-Aspire-5737Z NetworkManager[719]: <info>  [1619365146.7755] dhcp4 (wlp17s0): option dhcp_lease_time      => '3599'
Apr 25 11:39:06 python-Aspire-5737Z NetworkManager[719]: <info>  [1619365146.7755] dhcp4 (wlp17s0): option domain_name_servers  => '192.168.43.1'
Apr 25 11:39:06 python-Aspire-5737Z NetworkManager[719]: <info>  [1619365146.7756] dhcp4 (wlp17s0): option expiry               => '1619368745'
Apr 25 11:39:06 python-Aspire-5737Z NetworkManager[719]: <info>  [1619365146.7756] dhcp4 (wlp17s0): option host_name            => 'python-Aspire-5737Z'
Apr 25 11:39:06 python-Aspire-5737Z NetworkManager[719]: <info>  [1619365146.7756] dhcp4 (wlp17s0): option ip_address           => '192.168.43.221'
Apr 25 11:39:06 python-Aspire-5737Z NetworkManager[719]: <info>  [1619365146.7757] dhcp4 (wlp17s0): option next_server          => '192.168.43.1'
Apr 25 11:39:06 python-Aspire-5737Z NetworkManager[719]: <info>  [1619365146.7757] dhcp4 (wlp17s0): option requested_broadcast_address => '1'
Apr 25 11:39:06 python-Aspire-5737Z NetworkManager[719]: <info>  [1619365146.7757] dhcp4 (wlp17s0): option requested_domain_name => '1'
Apr 25 11:39:06 python-Aspire-5737Z NetworkManager[719]: <info>  [1619365146.7757] dhcp4 (wlp17s0): option requested_domain_name_servers => '1'
Apr 25 11:39:06 python-Aspire-5737Z NetworkManager[719]: <info>  [1619365146.7758] dhcp4 (wlp17s0): option requested_domain_search => '1'
Apr 25 11:39:06 python-Aspire-5737Z dbus-daemon[716]: [system] Activating via systemd: service name='org.freedesktop.nm_dispatcher' unit='dbus-org.freedesktop.nm-dispatcher.service' requested by ':1.12' (uid=0 pid=719 comm="/usr/sbin/NetworkManager --no-daemon " label="unconfined")
Apr 25 11:39:06 python-Aspire-5737Z NetworkManager[719]: <info>  [1619365146.7758] dhcp4 (wlp17s0): option requested_host_name  => '1'
Apr 25 11:39:06 python-Aspire-5737Z NetworkManager[719]: <info>  [1619365146.7758] dhcp4 (wlp17s0): option requested_interface_mtu => '1'
Apr 25 11:39:06 python-Aspire-5737Z NetworkManager[719]: <info>  [1619365146.7759] dhcp4 (wlp17s0): option requested_ms_classless_static_routes => '1'
Apr 25 11:39:06 python-Aspire-5737Z NetworkManager[719]: <info>  [1619365146.7759] dhcp4 (wlp17s0): option requested_nis_domain => '1'
Apr 25 11:39:06 python-Aspire-5737Z NetworkManager[719]: <info>  [1619365146.7759] dhcp4 (wlp17s0): option requested_nis_servers => '1'
Apr 25 11:39:06 python-Aspire-5737Z NetworkManager[719]: <info>  [1619365146.7759] dhcp4 (wlp17s0): option requested_ntp_servers => '1'
Apr 25 11:39:06 python-Aspire-5737Z NetworkManager[719]: <info>  [1619365146.7760] dhcp4 (wlp17s0): option requested_rfc3442_classless_static_routes => '1'
Apr 25 11:39:06 python-Aspire-5737Z NetworkManager[719]: <info>  [1619365146.7760] dhcp4 (wlp17s0): option requested_root_path  => '1'
Apr 25 11:39:06 python-Aspire-5737Z NetworkManager[719]: <info>  [1619365146.7760] dhcp4 (wlp17s0): option requested_routers    => '1'
Apr 25 11:39:06 python-Aspire-5737Z NetworkManager[719]: <info>  [1619365146.7761] dhcp4 (wlp17s0): option requested_static_routes => '1'
Apr 25 11:39:06 python-Aspire-5737Z NetworkManager[719]: <info>  [1619365146.7761] dhcp4 (wlp17s0): option requested_subnet_mask => '1'
Apr 25 11:39:06 python-Aspire-5737Z NetworkManager[719]: <info>  [1619365146.7761] dhcp4 (wlp17s0): option requested_time_offset => '1'
Apr 25 11:39:06 python-Aspire-5737Z NetworkManager[719]: <info>  [1619365146.7761] dhcp4 (wlp17s0): option requested_wpad       => '1'
Apr 25 11:39:06 python-Aspire-5737Z NetworkManager[719]: <info>  [1619365146.7762] dhcp4 (wlp17s0): option routers              => '192.168.43.1'
Apr 25 11:39:06 python-Aspire-5737Z NetworkManager[719]: <info>  [1619365146.7762] dhcp4 (wlp17s0): option subnet_mask          => '255.255.255.0'
Apr 25 11:39:06 python-Aspire-5737Z NetworkManager[719]: <info>  [1619365146.7762] dhcp4 (wlp17s0): state changed bound -> extended
Apr 25 11:39:06 python-Aspire-5737Z systemd[1]: Starting Network Manager Script Dispatcher Service...
Apr 25 11:39:06 python-Aspire-5737Z dbus-daemon[716]: [system] Successfully activated service 'org.freedesktop.nm_dispatcher'
Apr 25 11:39:06 python-Aspire-5737Z systemd[1]: Started Network Manager Script Dispatcher Service.
Apr 25 11:39:16 python-Aspire-5737Z systemd[1]: NetworkManager-dispatcher.service: Succeeded.

**Reply by killgapwalker** [reply]
do i need wpad ? :/

**Reply by Thermi** [reply]
No. WLAN has nothing to do with.

**Reply by killgapwalker** [reply]
Everything is requested but doesn't apply ... Where i can look all documentation about network manager and program language?

**Reply by Thermi** [reply]
What exactly are you asking? NetworkManager and any programming language have nothing to do with this issue.

**Reply by killgapwalker** [reply]
Im trying to understand which configuration need to be apply , but this wpad always show his finger in wireshark...

**Reply by Thermi** [reply]
I commented on the issue above. You first need to configure the right cipher proposals.
Wireshark is not useful for this problem. It doesn't show you more than strongSwan tells you in the logs.

**Reply by killgapwalker** [reply]
Same problem...
 ike=aes256gcm16-sha384-modp3072 cipher is right now...

**Reply by killgapwalker** [reply]
maybe ike=aes256gcm-sha2-384-modp3072 ???

**Reply by Thermi** [reply]
No. Those are invalid keywords. Try aes256gcm8- and aes256gcm12- instead of aes256cm16.

**Comment by killgapwalker** [comment]
Maybe i need to open radius server on my telephone ?

**Reply by Thermi** [reply]
No, stop guessing. It's either the IKE proposal, leftid, or right id. Just look for a working config on the Internet.
I found one that supposedly works on the nordvpn forums.

A modified version follows
```
conn NordVPN
  keyexchange=ikev2
  dpdaction=restart
  dpddelay=300s
  eap_identity="nord username (email address)"
  leftauth=eap-mschapv2
  leftsourceip=%config
  right=us699.nordvpn.com (I picked a server local to me)
  rightauth=pubkey
  rightsubnet=0.0.0.0/0
  rightid=%any
  type=tunnel
  auto=add
```

**Comment by killgapwalker** [comment]


**Reply by Thermi** [reply]
`Apr 26 11:41:35 python-Aspire-5737Z charon: 09[IKE] CHILD_SA same{1} established with SPIs c54153e0_i c3dd7a5e_o and TS 10.6.0.25/32 === 10.6.0.0/24`
You negotiated only for 10.6.0.0/24 as remote traffic selector. You need to set rightsubnet=0.0.0.0/0 to negotiate 0.0.0.0/0, which means any IPv4 address. Otherwise you will not be able to communicate with any IPv4 address that is outside of 10.6.0.0/24.

**Reply by killgapwalker** [reply]
im Reinstaled this client and downloaded only few things... somebody know how to enter in computer from porvider?

**Reply by Thermi** [reply]
Tell me what the exact problem is.

**Reply by killgapwalker** [reply]
Okay netstat -tupan is ok.
how to disallow ipv6 in ipsec ?
and which ports can disallow internet traffic in linux ?

**Reply by Thermi** [reply]
You can configure a local drop policy for any IPv6 address like this:
```
conn ipv6-drop
leftsubnet=::/0
rightsubnet=::/0
auto=route
type=drop
```

You should permit IPv6 NDP traffic and other ICMPv6 traffic though. Otherwise you will vanish from the local IPv6 network as if you were completely disconnected and hence probably loose all assigned IPV6 addresses due to timeouts.

If you don't want any IPv6 addresses on any interfaces at all, then you need to disable IPv6 for the interfaces you care about using `sysctl`. That is the `disable_ipv6` key.

You can restrict traffic using `iptables`, for example.

**Comment by killgapwalker** [comment]
Which ports can Block internet in udp, tcp...
can it be 636,53?

**Reply by Thermi** [reply]
What do you mean exactly? Do you mean in regards to browsing the WWW?

**Reply by killgapwalker** [reply]
I know that 22,23,137,138,139,140 ports can be hacked...
I suppose big range of ports need to be blocked , but nobody understand which of them block internet access to my computer...

**Reply by killgapwalker** [reply]
iptables -A INPUT -p tcp -m multiport --sport 1:65535 -m conntrack --ctstate NEW,ESTABLISHED -j DROP
iptables -A INPUT -p tcp -m multiport --dport 1:65535 -m conntrack --ctstate NEW,ESTABLISHED -j DROP
iptables -A INPUT -p udp -m multiport --sport 1:65535 -m conntrack --ctstate NEW,ESTABLISHED -j DROP
iptables -A INPUT -p udp -m multiport --dport 1:65535 -m conntrack --ctstate NEW,ESTABLISHED -j DROP
Why this rules doesn't block anything ?

**Reply by killgapwalker** [reply]
with -t filter and without this rules doesn't care...

**Reply by Thermi** [reply]
You are illinformed. None of these measures will help you anything because the primary attack vector is over your browser.
If you have a usable iptables configuration, no new connections are allowed from outside anyway. If you drop ctstate ESTABLISHED, then also packets of existing connections will be matched, regardless of which side initiated it.

iptables rules are also context sensitive, like any other firewall rule. The order they are in determines the actions taken.

Look at the output iptables-save to see the ruleset you are currently using.

**Reply by killgapwalker** [reply]
Okay , but there is a full internet of exploit which triger ports....
Im trying drop to understand do they actually work... 
How to write iptable which will filter all ports ?

**Reply by Thermi** [reply]
Just use the "rule hosts" sets from [here](https://github.com/QueuingKoala/netfilter-samples) and you're golden.

**Reply by killgapwalker** [reply]
Okay it's filtered ...
How can i open gre tunnel on this vpn connection from scratch of professional?

**Reply by Thermi** [reply]
Only with cooperation of the other side. The two sides involved need to be configured correctly. And they usually are not configured for several types of tunnels. Also the usage of GRE is usually not communicated via IKE explicitely but just, if at all, via putting the gre protocol number into the protocol field of the TS on both sides.
